src/main/web/common_ts/PhotoAlbumUploader.ts

```diff
@@ -77,6 +86,19 @@ class PhotoAlbumUploader{
 		}, {rootMargin: "-1px 0px 0px 0px", threshold: [1]}).observe(this.uploadOuterEl);
 	}
 
+	private onInputPaste(ev:ClipboardEvent):void{
+		if(!ge("photoAlbumUpload")){
+			// If we ended up on a different page than where the PhotoAlbumUploader was initially
+			// created, don't handle the current event and clean up the global event handlers.
+			document.body.removeEventListener("paste", PhotoAlbumUploader.pasteEventListener);
+			return;
+		}
```

Comment:
I don't like this, is there a way to attach a listener to some kind of navigation event so that this global state is reset on `ajaxNavigate`?

Reply:
That's a good question, there isn't, I'm going to make one.

For state variables, there's the `cur` global object that is reset on each ajax navigation.

Reply:
Added `ajaxNavCallbacks`, just an array of functions called before the page content is replaced